src/ledger/grainAllocation.js

```diff
@@ -225,25 +220,18 @@ function computeBalancedReceipts(
     }
   }
 
-  let totalPaid = G.ZERO;
-  const receipts = mapToArray(userUnderpayment, ([address, underpayment]) => {
-    const underpaymentProportion = G.toFloatRatio(
-      underpayment,
-      totalUnderpayment
-    );
-    const amount = G.multiplyFloat(budget, underpaymentProportion);
-    totalPaid = G.add(totalPaid, amount);
-    return {
-      address,
-      amount,
-    };
-  });
-  if (G.gt(totalPaid, budget)) {
-    throw new Error(
-      `invariant violation: paid ${totalPaid} greater than budget ${budget}`
-    );
-  }
-  return receipts;
+  const underpaymentTuples: [NodeAddressT, G.Grain][] = mapToArray(
+    userUnderpayment,
+    (x) => x
```

Comment:
Wondering if we can just have this as the default param for the mapToArray util. That way you can just do `mapToArray(userUnderpayment)`. Makes sense with the name of the function, its just turning it into an array.